lib/model/line_item.rb

```diff
@@ -17,6 +17,11 @@ def discounted?
       @discounted
     end
 
+    # CON: A good alternative here is a version where you don't need to mutate the original
+    # object. Of course it would require you to keep a reference to the original cart
+    # but it could be an interesting solution as well.
+    #
+    # For example it could make it easier to return a price comparsion in a future feature..
```

Comment:
You're right. I thought about immutability here, and in fact I tend to go for immutable objects more often than not, but the mutation in this case was quite isolated and made testing easier. 

However thinking about the use case you mentioned, having immutable objects would definitely help, so I'd change this.